include/sys/emmc.h

```diff
@@ -0,0 +1,303 @@
+                                          emmc_response_t *resp) {
+  device_t *idev = EMMC_METHOD_PROVIDER(dev, send_cmd);
+  return emmc_methods(idev->parent)->send_cmd(dev, cmd, arg1, arg2, resp);
+}
+
+/**
+ * \brief Send an app-specific e.MMC command
+ * \param dev e.MMC controller device
+ * \param cmd e.MMC command (described in JESD86-A441)
+ * \param arg1 first argument
+ * \param arg2 second argument
+ * \param exp_rt expected response type
+ * \param app_flags a set of flags defining how the controller should handle
+ *                  the command
+ * \param res pointer for response data to be written to or NULL
+ * \return EMMC_OK on success
+ */
+static inline emmc_result_t emmc_send_app_cmd(device_t *dev, uint32_t cmd,
+                                              uint32_t arg1, uint32_t arg2,
+                                              emmc_resp_type_t exp_rt,
+                                              emmc_app_flags_t app_flags,
+                                              emmc_response_t *resp) {
+  device_t *idev = EMMC_METHOD_PROVIDER(dev, send_app_cmd);
+  return emmc_methods(idev->parent)
+    ->send_app_cmd(dev, cmd, arg1, arg2, exp_rt, app_flags, resp);
+}
+
+/**
+ * \brief Wait until e.MMC signals
+ * \param dev e.MMC controller device
+ * \param wflags conditions to be met
+ */
+static inline int emmc_wait(device_t *dev, emmc_wait_flags_t wflags) {
+  device_t *idev = EMMC_METHOD_PROVIDER(dev, wait);
+  return emmc_methods(idev->parent)->wait(dev, wflags);
+}
+
+/**
+ * \brief Read data fetched by the controller
+ * \param dev e.MMC controller device
+ * \param buf pointer to where the data should be written to
+ * \param len expected data length
+ * \param read pointer for the number of read bytes or NULL
+ * \return EMMC_OK on success
+ */
+static inline emmc_result_t emmc_read(device_t *dev, void *buf, size_t len,
+                                      size_t *read) {
+  device_t *idev = EMMC_METHOD_PROVIDER(dev, read);
+  return emmc_methods(idev->parent)->read(dev, buf, len, read);
+}
+
+/**
+ * \brief Write data to the controller
+ * \param dev e.MMC controller device
+ * \param buf pointer to where the data should be read from
+ * \param len data length
+ * \param wrote pointer for the number of written bytes in or NULL
+ * \return EMMC_OK on success
+ */
+static inline emmc_result_t emmc_write(device_t *dev, const void *buf,
+                                       size_t len, size_t *wrote) {
+  device_t *idev = EMMC_METHOD_PROVIDER(dev, write);
+  return emmc_methods(idev->parent)->write(dev, buf, len, wrote);
+}
+
+/**
+ * \brief Geta a value describing the e.MMC device
+ * \param dev e.MMC controller device
+ * \param id value identifier
+ * \param var pointer to where the associated value should be written to
+ * \return -1 if value was fetched successfully, 0 if it wasn't
+ */
+static inline int emmc_get_prop(device_t *dev, emmc_prop_id_t id,
```

Comment:
Does "prop" stands for "property"? 
I guess you should put that word in comment above or change function name to feature whole word

Reply:
Yes. I;m keeping it short.